timeSeriesUtilities/info.json

```diff
@@ -0,0 +1,221 @@
+                        "content": ""
+                    },
+                    "content": ""
+                },
+                "content": ""
+            },
+            "content": ""
+        },
+        "content": ""
+    },
+    "icon_large_name": {
+        "name": {
+            "name": {
+                "name": {
+                    "name": {
+                        "name": "",
+                        "content": ""
+                    },
+                    "content": ""
+                },
+                "content": ""
+            },
+            "content": ""
+        },
+        "content": ""
+    },
+    "category": "",
+    "configuration": {},
+    "operations": [
+        {
+            "title": "Assemble Query Time Windows",
+            "operation": "assemble_query_time_windows",
+            "description": "Takes relative date, time period, date formatting data, plus optionally a list of existing datetimes and builds a list of time buckets from them spanning the time window specified by the relative date information.",
+            "parameters": [
+                {
+                    "title": "Relative Date",
+                    "type": "object",
+                    "name": "relativeDate",
+                    "required": true,
+                    "visible": true,
+                    "editable": true,
+                    "value": null,
+                    "tooltip": "JSON in the format {\"differenceType\": ..., \"differenceValue\": ...}"
+                },
+                {
+                    "title": "Period",
+                    "type": "select",
+                    "name": "period",
+                    "required": true,
+                    "visible": true,
+                    "editable": true,
+                    "value": null,
+                    "options": [
+                        "Hourly",
+                        "Daily",
+                        "Weekly",
+                        "Monthly",
+                        "Quarterly",
+                        "Yearly"
+                    ]
+                },
+                {
+                    "title": "Date Format String",
+                    "type": "text",
+                    "name": "dateFormatString",
+                    "required": true,
+                    "visible": true,
+                    "editable": true,
+                    "value": "YYYY-MM-DDTHH:mm:ss.SSS[Z]",
+                    "tooltip": "How dates are expected to be formatted in the chart JSON"
+                },
+                {
+                    "title": "Existing Times",
+                    "type": "object",
+                    "name": "existingTimes",
+                    "required": false,
+                    "visible": true,
+                    "editable": true,
+                    "value": null,
+                    "tooltip": "The list of times currently shown in the chart, if the chart has previously been generated"
```

Comment:
list of objects are expected here?

Reply:
Yes, this is expected to be a list, hence the "object" parameter type. The way C3 charts of this type are formatted, the x axis values are a list of date time values. So when adding to an existing chart we take in that list from the extant chart and add any needed additional times to it